File: work@tech/Pastries.java
package com.practice;

import java.util.Scanner;

public class Pastries {
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		int value[] = new int[2];

		value[0] = sc.nextInt();
		value[1] = sc.nextInt();

		int pastries = value[0];
		int queue = value[1];
		int order[] = new int[queue];

		for (int i = 0; i < queue; i++) {
			order[i] = sc.nextInt();
			int remainingPastries = pastries-order[i];
			if (remainingPastries >= -1){
				System.out.println("Enjoy your dessert!");
			}else {
				System.out.println("Sorry, we are all out!");
			}

			pastries = remainingPastries;
		}
	}
}

File: c++work@tech/Pastries.cpp
//
// Created by shawm on 26-09-2022.
//
#include <iostream>
using namespace std;

void pastries(int n, int *a, int p)
{
    for(int i = 0; i < n; i++)
    {
        if(p>0) {
            p = p - a[i];
            cout<<"Enjoy your dessert!"<<endl;
        }
        else{
            cout<<"Sorry, we are all out!"<<endl;
        }

    }
}
int main()
{
    int noOfPastries, noOfcustomers;
    cin>>noOfPastries>>noOfcustomers;
    int arr[noOfcustomers];
    for(int i = 0; i < noOfcustomers; i++)
    {
        cin>>arr[i];
    }

    pastries(noOfcustomers, arr, noOfPastries);





}